servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpExecutionStrategy.java

```diff
@@ -285,13 +250,12 @@ private static byte generateOffloadsFlag(final HttpExecutionStrategy strategy) {
 
     private Executor executor(final Executor fallback) {
         requireNonNull(fallback);
-        assert !threadAffinity : "Thread affinity not supported";
         return executor == null ? fallback : executor;
     }
 
     // Visible for testing
     boolean hasThreadAffinity() {
```

Comment:
remove this method (only used in a single test, just kill the state that uses it too)?